src/Test/Perf/util/test_util.csx

```diff
@@ -0,0 +1,239 @@
+#r "System.IO.Compression.FileSystem"
+
+using System.IO;
+using System.IO.Compression;
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Net;
+
+//
+// Directory Locating Functions
+//
+
+string _myWorkingFile = null;
+
+void InitUtilities([CallerFilePath] string sourceFilePath = "")
+{
+    _myWorkingFile = sourceFilePath;
+}
+
+/// Returns the directory that houses the currenly executing script.
+string MyWorkingDirectory()
+{
+    if (_myWorkingFile == null)
+    {
+        throw new Exception("Tests must call InitUtilities before doing any path-dependent operations.");
+    }
+    return Directory.GetParent(_myWorkingFile).FullName;
+}
+
+/// Returns the directory that you can put artifacts like
+/// etl traces or compiled binaries
+string MyArtifactsDirectory()
+{
+    var path = Path.Combine(MyWorkingDirectory(), "artifacts");
+    Directory.CreateDirectory(path);
+    return path;
+}
+
+string MyTempDirectory()
+{
+    var workingDir = MyWorkingDirectory();
+    var path = Path.Combine(workingDir, "temp");
+    Directory.CreateDirectory(path);
+    return path;
+}
+
+string RoslynDirectory()
+{
+    var workingDir = MyWorkingDirectory();
+    var srcTestPerf = Path.Combine("src", "Test", "Perf").ToString();
+    return workingDir.Substring(0, workingDir.IndexOf(srcTestPerf));
+}
+
+string BinDebugDirectory()
+{
+    return Path.Combine(RoslynDirectory(), "Binaries", "Debug");
+}
+
+string BinReleaseDirectory()
+{
+    return Path.Combine(RoslynDirectory(), "Binaries", "Release");
+}
+
+string DebugCscPath()
+{
+    return Path.Combine(BinDebugDirectory(), "csc.exe");
+}
+
+string ReleaseCscPath()
+{
+    return Path.Combine(BinReleaseDirectory(), "csc.exe");
+}
+
+string DebugVbcPath()
+{
+    return Path.Combine(BinDebugDirectory(), "vbc.exe");
+}
+
+string ReleaseVbcPath()
+{
+    return Path.Combine(BinReleaseDirectory(), "vbc.exe");
+}
+
+//
+// Process spawning and error handling.
+//
+
+class ProcessResult
+{
+    public string ExecutablePath {get; set;}
+    public string Args {get; set;}
+    public int Code {get; set;}
+    public string StdOut {get; set;}
+    public string StdErr {get; set;}
+
+    public bool Failed => Code != 0;
```

Comment:
If I want to see if a process failed, `if(result.Failed)` is easier to read than `if(!result.Succeeded)`.  

Not strictly necessary, but a readability improvement I think.